x/ecocredit/server/basket/query_basket_test.go

```diff
@@ -34,3 +34,31 @@ func TestKeeper_Basket(t *testing.T) {
 	})
 	require.Error(t, err)
 }
+
+func TestKeeper_BasketClasses(t *testing.T) {
+	t.Parallel()
+	s := setupBase(t)
+
+	// add a basket
+	basketDenom := "foo"
+	err := s.stateStore.BasketStore().Insert(s.ctx, &basketv1.Basket{
+		BasketDenom: basketDenom,
+	})
+	require.NoError(t, err)
+
+	// add a basket class
+	classId := "C01"
+	err = s.stateStore.BasketClassStore().Insert(s.ctx, &basketv1.BasketClass{
+		BasketId: 1,
+		ClassId:  classId,
+	})
+	require.NoError(t, err)
+
+	// query
+	res, err := s.k.Basket(s.ctx, &baskettypes.QueryBasketRequest{
+		BasketDenom: basketDenom,
+	})
+	require.NoError(t, err)
+	require.Equal(t, basketDenom, res.Basket.BasketDenom)
+	require.Equal(t, []string{classId}, res.Classes)
+}
```

Comment:
I think we can merge the 2 tests above - no need to duplicate most o the code.

Also, ideally would be great to make a test on the RPC server level (without running the server). We can do it in a separate PR.